File: KaraokeMax/Services/Banco de Dados/UsuarioService.cs
using System;
using System.Collections.Generic;
using System.Text.Json;
using System.Threading.Tasks;
using System.Windows;
using KaraokeMax.Models;
using MySql.Data.MySqlClient;

namespace KaraokeMax.Services.Banco_de_Dados
{
    static class UsuarioService
    {
        private static string connectionString = "server=mysql-karaokemax.alwaysdata.net;database=karaokemax_db;uid=430017;pwd=KaraokePucSecreto;";
        public static void CriarUsuario(String nome, String email, String tipo)
        {
            try
            {
                using (var connection = new MySqlConnection(connectionString))
                {
                    connection.Open();
                    string query = "INSERT INTO Usuarios (nome, email, tipo) VALUES (@nome, @email, @tipo)";
                    using (var command = new MySqlCommand(query, connection))
                    {
                        command.Parameters.AddWithValue("@nome", nome);
                        command.Parameters.AddWithValue("@email", email);
                        command.Parameters.AddWithValue("@tipo", tipo);
                        command.ExecuteNonQuery();
                    }
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show("Erro ao criar usuário: " + ex.Message);
                throw;
            }
        }

        public static void AlterarSenhaPrimeiroAcesso(string id, string novaSenha)
        {
            try
            {
                using (var connection = new MySqlConnection(connectionString))
                {
                    connection.Open();
                    string query = "UPDATE Usuarios SET senha = @senha, primeiroAcesso = false WHERE id = @id";
                    using (var command = new MySqlCommand(query, connection))
                    {
                        command.Parameters.AddWithValue("@senha", novaSenha);
                        command.Parameters.AddWithValue("@id", id);
                        command.ExecuteNonQuery();
                    }
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show("Erro ao alterar senha: " + ex.Message);
                throw;
            }
        }

        public static UsuarioModel VerificaLogin(string email, string senha)
        {
            try
            {
                using (var connection = new MySqlConnection(connectionString))
                {
                    connection.Open();
                    string query = "SELECT * FROM Usuarios WHERE email = @email AND senha = @senha";
                    using (var command = new MySqlCommand(query, connection))
                    {
                        command.Parameters.AddWithValue("@email", email);
                        command.Parameters.AddWithValue("@senha", senha);
                        using (var reader = command.ExecuteReader())
                        {
                            if (reader.Read())
                            {
                                return new UsuarioModel(
                                    reader["id"].ToString(),
                                    Convert.ToBoolean(reader["primeiroAcesso"]),
                                    reader["nome"].ToString(),
                                    reader["email"].ToString(),
                                    reader["senha"].ToString(),
                                    reader["tipo"].ToString()
                                );
                            }
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show("Erro ao buscar usuário: " + ex.Message);
                throw;
            }
            return null;
        }

        public static void CriaSenha(String idUsuario, String novaSenha)
        {
            try
            {
                using (var connection = new MySqlConnection(connectionString))
                {
                    connection.Open();
                    string query = "UPDATE Usuarios SET senha = @senha, primeiroAcesso = false WHERE id = @id";
                    using (var command = new MySqlCommand(query, connection))
                    {
                        command.Parameters.AddWithValue("@senha", novaSenha);
                        command.Parameters.AddWithValue("@id", idUsuario);
                        command.ExecuteNonQuery();
                    }
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show("Erro ao criar senha: " + ex.Message);
                throw;
            }
        }

        public static void DeletarUsuario(string id)
        {
            try
            {
                using (var connection = new MySqlConnection(connectionString))
                {
                    connection.Open();
                    string query = "DELETE FROM Usuarios WHERE id = @id";
                    using (var command = new MySqlCommand(query, connection))
                    {
                        command.Parameters.AddWithValue("@id", id);
                        command.ExecuteNonQuery();
                    }
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show("Erro ao deletar usuário: " + ex.Message);
                throw;
            }
        }
    }
}

File: Karaoke_Backend/Services/Usuario-Service.js
const Usuario = require('../Models/Usuario-Model');

async function criarUsuario(nome, email, tipo) {
	return await Usuario.create({ nome, email, tipo });
}

async function alterarSenhaPrimeiroAcesso(id, novaSenha) {
	return await Usuario.update(
		{ senha: novaSenha, primeiroAcesso: false },
		{ where: { id } }
	);
}

async function verificaLogin(email, senha) {
	const usuario = await Usuario.findOne({ where: { email, senha } });
	if (usuario) {
		return usuario;
	}
	throw new Error('Usuário não encontrado');
}

async function criaSenha(idUsuario, novaSenha) {
	return await Usuario.update(
		{ senha: novaSenha, primeiroAcesso: false },
		{ where: { id: idUsuario } }
	);
}

async function deletarUsuario(id) {
	return await Usuario.destroy({ where: { id } });
}

async function buscarTodosUsuarios() {
	return await Usuario.findAll();
}

async function buscarPorId(id) {
	const usuario = await Usuario.findByPk(id, {
		include: [{
			association: 'notificacoes',
			where: { lida: false },
			required: false
		}]
	});
	if (usuario) {
		return usuario;
	}
	throw new Error('Usuário não encontrado');
}

async function atualizarTokenUsuario(id, token) {
	return await Usuario.update(
		{ token },
		{ where: { id } }
	);
}

module.exports = {
	criarUsuario,
	alterarSenhaPrimeiroAcesso,
	verificaLogin,
	criaSenha,
	deletarUsuario,
	buscarTodosUsuarios,
	buscarPorId,
	atualizarTokenUsuario,
};
